Extension/readme.developer.md

```diff
@@ -0,0 +1,337 @@
+# Developer Documentation
+
+## Table of Contents
+- [Table of Contents](#table-of-contents)
+- [Setup](#setup)
+    - [Required Tools](#required-tools)
+    - [Setting up the repository](#setting-up-the-repository)
+- [Building the Extension](#building-the-extension)
+    - [When using F5 Debug](#when-using-f5-debug)
+    - [From Inside VS Code](#from-inside-vs-code)
+    - [From Command line](#from-command-line)
+- [Use of an isolated VS Code environment](#use-of-an-isolated-vs-code-environment)
+- [Testing](#testing)
+    - [Unit tests](#unit-tests)
+    - [Scenario Tests](#scenario-tests)
+- [Scripts](#scripts)
+    - [`yarn scripts`](#yarn-scripts)
+    - [`yarn show`](#yarn-show)
+    - [`yarn clean`](#yarn-clean)
+    - [`yarn test`](#yarn-test)
+    - [`yarn code`](#yarn-code)
+    - [`yarn generate-native-strings`](#yarn-generate-native-strings)
+    - [`yarn generate-options-schema`](#yarn-generate-options-schema)
+    - [`yarn copy-walkthrough-media`](#yarn-copy-walkthrough-media)
+    - [`yarn prep`](#yarn-prep)
+    - [`yarn lint`](#yarn-lint)
+    - [`yarn compile`](#yarn-compile)
+    - [`yarn watch`](#yarn-watch)
+    - [`yarn verify`](#yarn-verify)
+    - [`yarn webpack`](#yarn-webpack)
+    - [`yarn postinstall`](#yarn-postinstall)
+
+## Setup
+
+### Required Tools 
+
+* [Node.js](https://nodejs.org/en/download/) v16.*
+* Yarn - use `npm install -g yarn` to install
+
+### Setting up the repository
+
+`git clone https://github.com/microsoft/vscode-cpptools.git`
+
+`yarn install`
```

Comment:
I noticed that after one of your recent changes, I needed re-run `yarn install` -- can you add some note about when users may need to re-run `yarn install`? Previously, running debug would auto-run yarn install, so I'm worried about people not being aware of this change in process.

Reply:
Under normal circumstances, you only need to run `yarn install` when package dependencies or the `yarn.lock` file changes.

If you are switching between branches, you should run `yarn clean` and `yarn install`.

Reply:
added a note in the file